modules/frontend/tag_sharder.go

```diff
@@ -257,7 +257,9 @@ func (s searchTagSharder) backendRequests(ctx context.Context, tenantID string,
 	// get block metadata of blocks in start, end duration
 	startT := time.Unix(int64(start), 0)
 	endT := time.Unix(int64(end), 0)
-	blocks := blockMetasForSearch(s.reader.BlockMetas(tenantID), startT, endT, backend.DefaultReplicationFactor)
+	blocks := blockMetasForSearch(s.reader.BlockMetas(tenantID), startT, endT, func(m *backend.BlockMeta) bool {
+		return m.ReplicationFactor == backend.DefaultReplicationFactor
```

Comment:
Can we not use the same logic here as in the search sharder?

Reply:
Actually, we need to. Otherwise these requests will fail when we stop flushing ingester blocks. Great catch!